src/live/src/user-to-user/live-service-facade.js

```diff
@@ -0,0 +1,58 @@
+import PubNub from 'pubnub';
+
+import Client from '../../../client';
+import { User } from '../../../entity';
+import { PubNubListener } from './pubnub-listener';
+import { getLiveService } from './live-service';
+
+
+function _getLiveService() {
+  const client = Client.sharedInstance();
+  return getLiveService(client);
+}
+
+function register() {
+  const liveService = _getLiveService();
+  const activeUser = User.getActiveUser();
+
+  return liveService.registerUser(activeUser)
+    .then((pubnubConfig) => {
+      const pubnubClient = new PubNub(pubnubConfig);
+      const listener = new PubNubListener();
+      liveService.initialize(pubnubClient, listener);
+    });
+}
+
+function unregister() {
+  const liveService = _getLiveService();
+  return liveService.unregisterUser()
+    .then((resp) => {
+      liveService.uninitialize();
+      return resp;
+    });
+}
+
+function onConnectionStatusUpdates(func) {
+  _getLiveService().onConnectionStatusUpdates(func);
+}
+
+function offConnectionStatusUpdates(func) {
+  _getLiveService().offConnectionStatusUpdates(func);
+}
+
+function unsubscribeFromAll() {
+  _getLiveService().unsubscribeFromAll();
+}
+
+function isInitialized() {
```

Comment:
I think from a user's standpoint, `LiveServiceFacade.register()` (and I am really not a fan of the "register" word here) is the initialization. The requirement to do an initialization (register for live service) will be known to them, this has been decided. And having the ability to check whether this has been done or not, at any point in time, seems valuable to me - if I am using the SDK, I would like to have that. If we have an initialization step (the "registration"), I think we should have a way to tell if it has been done. I don't think it tells them about PubNub, necessarily. It's an initialization step and they are going to know it exists, so might as well have a check for it.

Registration rant coming.. :)

This brings me back to the fact that I don't like the "register user for live service" part of the vocabulary. I see it more as a "login" of sorts, rather than a registration. An authentication step. You call an endpoint, provide credentials and get credentials for PubNub. It shouldn't be called "registration" for users who don't use the SDK (and the endpoint is /register-realtime - when we now call it "live service").

In the SDKs, the more I think about it, the more I convince myself this shouldn't be up to the user at all - we should do it implicitly (like we request permissions to publish/subscribe). The user doesn't care about "registering for live service" (whatever this would mean to them). Thy care about sending and receiving messages. The fact we choose to use PubNub and have an authentication step, should only be a concern for us, if user is using an SDK. I was very much in favour of this, before I "submerged" in the current state and totally forgot about it :) But you reminded me with this comment. @tejasranade If there's time, I'd like to do that. But I doubt there is, especially if it would need to be implemented in other SDKs too. But I want to have it out there, for the record :D